.github/workflows/ci.yml

```diff
@@ -53,9 +53,9 @@ jobs:
           coverage run --source=npe2 -m pytest --color yes
 
       - name: Upload coverage as artifact
-        uses: actions/upload-artifact@v3
+        uses: actions/upload-artifact@v4
         with:
-          name: coverage reports
+          name: coverage reports ${{ matrix.platform }} py ${{ matrix.python-version }} ${{ matrix.pydantic }}
```

Comment:
@Czaki we can't use ` <` in the coverage report name, so just using `${{ matrix.pydantic }}` is not an option. Can we replace the `<2` and `>2` somehow by editing the `${{ matrix.pydantic }}` bit? Do you have other ideas?